controls/datagrid/search-as-you-type.md

```diff
@@ -0,0 +1,82 @@
+
+You can check how the Search Panel UI looks on Windows at the image below:
+
+![Telerik .NET MAUI DataGrid Search Panel](images/datagrid-search-searchpanel.png)
+
+### Search Settings
+
+The default searching behavior works in the following way - all the words divided by space are trimmed from the search text and searched by according to the `LogicalOperator` property.
+
+Below you can find a list of the available configuration options applied through the `RadDataGrid`.`SearchSettings`:
+
+* `IntermediateSearchText`(`string`)&mdash;Specifies the text of the search entry. Change of this value will not necessarily trigger a search, i.e. it will not necessarily change the `SearchText` property. The `SearchText` property is updated in accordance `SearchTrigger` property.
+* `SearchText`(`string`)&mdash;the search text. Before a search is started, this value is trimmed, and the `SearchStarting` event is raised. `SearchStarting` event allows to change the effective search text that will be used for searching.
+* `SearchTrigger`(`DataGridSearchTrigger`)&mdash;Indicates when a search operation should be performed while the end user is typing in the search entry of the search panel. Available options are:
+    * `TextChanged`&mdash;a search operation is triggered every time the text of the entry changes.
+    * `Completed`&mdash;a search operation is triggered when the corresponding entry completes (by pressing Enter/Return key).
+* `TextMatchMode`(`DataGridSearchTextMatchMode`)&mdash;Indicates how a search term should be searched for within a text. Available options are:
+    * `Contains`&mdash;an item is marked as a search-match as long as the search text is contained anywhere in the text element (cell or other).
+    * `StartsWith`&mdash;an item is marked as a search-match only if the text of the text element (cell or other) contains a word that starts with the search text.
+    * `WholeWord`&mdash;an item is marked as a search-match only if the text of the text element (cell or other) contains a word that matches the search text.
+* `LogicalOperator`(`Telerik.Maui.LogicalOperator`)&mdash;Specifies the logical operator when the search text is comprised of more than one search term.
+* `CaseSensitive`(`bool`)&mdash;Indicates whether string comparison should be case sensitive.
+* `ApplyFilter`(`bool`)&mdash; Determines whether business items that do not satisfy the search criteria should be filtered.
+* `SearchMatchesCount`(`int`)&mdash;Gets the number of search matches.
+* `ProvideSearchMatchesAction`(`Action<DataGridSearchProbe>`)&mdash;Applies custom action that is invoked when determining the search matches for an item. Use this action when you need to specify search matches based on custom logic.
+
+### Events
```

Comment:
* Try adding some context.
* Move the event out of the list as we generally don't use single-item lists.
```suggestion
## Events

To further customize the Search as You Type functionality, use the `SearchStarting` event&mdash;Raised before searching starts. The effective search text can be changed and searching can be cancelled from the arguments of the event:
```